extensions/data-plane/data-plane-api/src/main/java/org/eclipse/dataspaceconnector/dataplane/api/controller/DataPlanePublicApiController.java

```diff
@@ -158,17 +160,49 @@ private Response handle(ContainerRequestContext requestContext) {
     /**
      * Create a {@link DataFlowRequest} based on the decoded claim token and the request content.
      */
-    private DataFlowRequest createDataFlowRequest(ClaimToken claims, Map<String, String> properties) {
-        var dataAddress = typeManager.readValue(claims.getClaims().get(DataPlaneConstants.DATA_ADDRESS), DataAddress.class);
+    private DataFlowRequest createDataFlowRequest(ClaimToken claims, ContainerRequestContext requestContext) {
+        var origDataAddress = typeManager.readValue(claims.getClaims().get(DataPlaneConstants.DATA_ADDRESS), DataAddress.class);
+        if (!HttpDataAddressSchema.TYPE.equals(origDataAddress.getType())) {
```

Comment:
Seems there is a misunderstanding here. Public DPF API is not restricted to accessing Rest API. 
It can be used to interrogate a database, a store...

Reply:
ok, so I need to make it conditional, i.e. just update the Data Address if it's of type `HttpDataAddressSchema.TYPE`
Does not make sense to pass path and query params if it's pointing to a SQL database.

Reply:
Some can still be useful, for example the query parameters if you want to target a DB.
This is why I would be in favor of puting there info anyway in the DataFlowRequest properties, and then in the factory (e.g. HttpDataSourceFactory) that is called, you choise whether you want to use these properties to complete the data address of not.